File: Selenium/Project/Activity7.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
import time
from selenium.webdriver import ActionChains

with webdriver.Firefox() as driver:
    driver.get("http://alchemy.hguy.co/crm")
    driver.maximize_window()
    action = ActionChains(driver)
    wait = WebDriverWait(driver, 10)

    user_name_input = driver.find_element(By.ID, "user_name")
    password_input = driver.find_element(By.ID, "username_password")
    login_button = driver.find_element(By.ID, "bigbutton")
    user_name_input.send_keys("admin")
    password_input.send_keys("pa$$w0rd")
    login_button.click()
    time.sleep(5)
    actions_button_home_page = driver.find_element(By.ID, "tab-actions")
    wait.until(EC.visibility_of(actions_button_home_page))

    sales_menu_item = driver.find_element(By.XPATH, "//a[text() = 'Sales']")
    leads = driver.find_element(By.XPATH, "//a[text() = 'Leads']")
    action.move_to_element(sales_menu_item).pause(3).click(leads).pause(5).perform()

    wait.until(EC.visibility_of_element_located((By.CSS_SELECTOR, ".table-responsive")))

    additional_info = driver.find_element(By.XPATH, "//div[@class= 'list-view-rounded-corners']/table/tbody/tr[1]//span[@title = 'Additional Details']")
    additional_info.click()
    time.sleep(5)

    phone_number = driver.find_element(By.XPATH, "//h2[contains(text(), 'Lead:')]/following-sibling::span").text
    print("Phone number: " +phone_number)

File: Selenium/Project/Activity7.java
package project;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Reporter;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

public class Activity7 {

	public WebDriver driver;
	public WebDriverWait wait;

	@BeforeClass
	public void setup() {
		driver = new FirefoxDriver();
		wait = new WebDriverWait(driver, Duration.ofSeconds(10));
		driver.get("http://alchemy.hguy.co/crm");
		driver.manage().window().maximize();
	}

	@Test
	public void testActivity7() throws InterruptedException {
		WebElement userNameInput = driver.findElement(By.id("user_name"));
		WebElement passwordInput = driver.findElement(By.id("username_password"));
		WebElement loginButton = driver.findElement(By.id("bigbutton"));
		userNameInput.sendKeys("admin");
		passwordInput.sendKeys("pa$$w0rd");
		loginButton.click();
		Thread.sleep(Duration.ofSeconds(5));
		WebElement actionsButtonHomePage = driver.findElement(By.id("tab-actions"));
		wait.until(ExpectedConditions.visibilityOf(actionsButtonHomePage));

		Actions action = new Actions(driver);
		WebElement salesMenuItem = driver.findElement(By.xpath("//a[text() = 'Sales']"));
		WebElement leads = driver.findElement(By.xpath("//a[text() = 'Leads']"));
		action.moveToElement(salesMenuItem).pause(Duration.ofSeconds(3)).click(leads).pause(Duration.ofSeconds(5)).build().perform();

		wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector(".table-responsive")));

		WebElement additionalInfo = driver.findElement(By.xpath("//div[@class= 'list-view-rounded-corners']/table/tbody/tr[1]//span[@title = 'Additional Details']"));

		additionalInfo.click();
		Thread.sleep(Duration.ofSeconds(5));



		String phoneNumber = driver.findElement(By.
				xpath("//h2[contains(text(), 'Lead:')]/following-sibling::span")).getText();

		System.out.println("Phone number: " +phoneNumber);
		Reporter.log("Phone number: " +phoneNumber);

	}
	@AfterClass
	public void teardown() {
		driver.quit();
	}
}
